pkg/skaffold/config/portforward.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"strconv"
+)
+
+const (
+	compat   = "compat"
+	user     = "user"
+	services = "services"
+	debug    = "debug"
+	pods     = "pods"
+	off      = "off"
+)
+
+// PortForwardOptions are options set by the command line for port forwarding
+// with additional configuration information as well
+type PortForwardOptions struct {
+	Modes []string
+}
+
+// Enabled checks if the port-forwarding options indicates that forwarding should be enabled.
+func (p PortForwardOptions) Enabled() bool {
+	if len(p.Modes) == 0 {
+		return false
+	}
+	// --port-forward was previously a boolean, so accept pflag's boolean values.
+	// This method accepts "none" or bool variants to be mixed with others,
+	// leaving complaining to be done by Validate.
+	for _, o := range p.Modes {
+		b, err := strconv.ParseBool(o)
+		if err == nil && !b {
+			return false
+		}
+		if o == off {
+			return false
+		}
+	}
+	return true
+}
+
+// Validate checks that the port-forward options are ok.
+// For example, `none` is not mixed with other values.
+func (p PortForwardOptions) Validate() error {
+	// boolean values (true/false/1/0), `compat`, and `none` must be used alone
+	for _, o := range p.Modes {
+		if _, err := strconv.ParseBool(o); err == nil {
+			if len(p.Modes) > 1 {
+				return fmt.Errorf("port-forward %q cannot be combined with other options", o)
+			}
+		} else {
+			switch o {
+			case off, compat:
```

Comment:
"compat" being a valid value is not mentioned in the help. Shd we even allow it here? After scrolling to the rest of the code feels like its used as an indication for backward compatibility.

Reply:
It's not intended to be used from the command-line — it's there purely for the NoOptDefValue (when `--port-forward` is used with no modes).  I'll make this clearer with a comment on PortForwardOptions.